### DISCUSSION TITLE
upload txt file using pynetbox for plagin netbox-document

### DISCUSSION THREAD

**Discussion by kaoplx** [netbox-community/pynetbox#606] [General]
I want upload txt for site.
`
f = open('file.txt', 'rb'). read()
f = base64.b64encode(f)
nb.plagins.documents.site_documents.create(site='11', name='doc', document_type='other', document= f)
`
Take error - object of type bytes is not JSON serializable.

If
` f = open('file.txt', 'rb'). read()`
Error - object has no attribute "JSON"

**Comment by squintfox** [comment]
@kaoplx A Base64FileField expects f to be a string, not a bytes-like object.  Try this:

```f = base64.b64encode(f).decode('utf-8')```

Also, this is probably best raised in https://github.com/jasonyates/netbox-documents, rather than here.
